### PR TITLE
iperf3: updated versions from 3.6 to 3.16

### LINKED ISSUE #44096: Update version of iperf3 from 3.6 to 3.16
### Steps to reproduce

Spack currently only explicitly has version 3.6 of the iPerf3 package (out of ESnet / LBNL). This makes the default the latest version of 3.16, and adds some other versions (found in some Linux distros, for possible compat purposes).

```
diff --git a/var/spack/repos/builtin/packages/iperf3/package.py b/var/spack/repos/builtin/packages/iperf3/package.py
index 162ea1b65a..aeca3aafcf 100644
--- a/var/spack/repos/builtin/packages/iperf3/package.py
+++ b/var/spack/repos/builtin/packages/iperf3/package.py
@@ -12,6 +12,10 @@ class Iperf3(AutotoolsPackage):
     maintained project."""
 
     homepage = "https://software.es.net/iperf/"
-    url = "https://github.com/esnet/iperf/archive/3.6.tar.gz"
+    url = "https://downloads.es.net/pub/iperf/iperf-3.16.tar.gz"
 
+    version("3.16", sha256="cc740c6bbea104398cc3e466befc515a25896ec85e44a662d5f4a767b9cf713e")
+    version("3.14", sha256="723fcc430a027bc6952628fa2a3ac77584a1d0bd328275e573fc9b206c155004")
+    version("3.12", sha256="72034ecfb6a7d6d67e384e19fb6efff3236ca4f7ed4c518d7db649c447e1ffd6")
+    version("3.9", sha256="24b63a26382325f759f11d421779a937b63ca1bc17c44587d2fcfedab60ac038")
     version("3.6", sha256="1ad23f70a8eb4b892a3cbb247cafa956e0f5c7d8b8601b1d9c8031c2a806f23f")
```

### Error message

_No response_

### Information on your system

* **Spack:** 0.21.0 (65d3221a9c436e77af9e802e543c71bd702ff4e2)
* **Python:** 3.10.12
* **Platform:** linux-ubuntu22.04-cascadelake
* **Concretizer:** clingo

### General information

- [X] I have run `spack debug report` and reported the version of Spack/Python/Platform
- [X] I have searched the issues of this repo and believe this is not a duplicate
- [X] I have run the failing commands in debug mode and reported the output

### PR DESCRIPTION
**Pull request by dmagdavector**
Spack currently only explicitly has version 3.6 of the iPerf3 package (out of ESnet / LBNL). This makes the default the latest version of 3.16, and adds some other versions (found in some Linux distros, for possible compatibility purposes).

closes #44096

### CODE REVIEW

File: var/spack/repos/builtin/packages/iperf3/package.py [hunk 1]
```diff
@@ -12,8 +12,12 @@ class Iperf3(AutotoolsPackage):
...
-    url = "https://github.com/esnet/iperf/archive/3.6.tar.gz"
+    url = "https://downloads.es.net/pub/iperf/iperf-3.16.tar.gz"
 
     license("BSD-3-Clause-LBNL")
 
+    version("3.16", sha256="cc740c6bbea104398cc3e466befc515a25896ec85e44a662d5f4a767b9cf713e")
+    version("3.14", sha256="723fcc430a027bc6952628fa2a3ac77584a1d0bd328275e573fc9b206c155004")
+    version("3.12", sha256="72034ecfb6a7d6d67e384e19fb6efff3236ca4f7ed4c518d7db649c447e1ffd6")
+    version("3.9", sha256="24b63a26382325f759f11d421779a937b63ca1bc17c44587d2fcfedab60ac038")
     version("3.6", sha256="1ad23f70a8eb4b892a3cbb247cafa956e0f5c7d8b8601b1d9c8031c2a806f23f")
```
Comment (wdconinc) [inline var/spack/repos/builtin/packages/iperf3/package.py]: Change in URL causes change in sha256, please verify that this is indeed the new checksum:
```suggestion
    version("3.6", sha256="de5d51e46dc460cc590fb4d44f95e7cad54b74fea1eba7d6ebd6f8887d75946e")
```

File: var/spack/repos/builtin/packages/iperf3/package.py [hunk 2]
```diff
@@ -12,8 +12,12 @@ class Iperf3(AutotoolsPackage):
     maintained project."""
 
     homepage = "https://software.es.net/iperf/"
-    url = "https://github.com/esnet/iperf/archive/3.6.tar.gz"
+    url = "https://downloads.es.net/pub/iperf/iperf-3.16.tar.gz"
 
     license("BSD-3-Clause-LBNL")
 
+    version("3.16", sha256="cc740c6bbea104398cc3e466befc515a25896ec85e44a662d5f4a767b9cf713e")
```
Comment (wdconinc) [inline var/spack/repos/builtin/packages/iperf3/package.py:20]: There is now also 3.17 as of Friday.

File: var/spack/repos/builtin/packages/iperf3/package.py (hunk 1 above)
Comment (dmagdavector) [reply var/spack/repos/builtin/packages/iperf3/package.py]: Yes, the "de5…" is the now-correct hash with the new URL. See also:
 * http://software.es.net/iperf/news.html#iperf-3-6-released
 
 (Had two working areas for the code (client/server) and pushed from the incorrect one.)

File: var/spack/repos/builtin/packages/iperf3/package.py (hunk 2 above)
Comment (dmagdavector) [reply var/spack/repos/builtin/packages/iperf3/package.py:20]: Happy for Spack to default to 3.17. SHA256 is 
* `077ede831b11b733ecf8b273abd97f9630fd7448d3ec1eaa789f396d82c8c943`

as per:

* http://software.es.net/iperf/news.html#iperf-3-17-released

**Comment by wdconinc** [comment]
Please feel free to re-request a review when the new version and hash are here. I don't see any other issues.

**Review by wdconinc — DISMISSED** [review]
Looks good. Thanks for the contribution!

**Comment by wdconinc** [comment]
Could you file the protobuf change as a separate PR please?

**Comment by dmagdavector** [comment]
> Could you file the protobuf change as a separate PR please?

Mis-click / mis-merge on my part: that was my original intention: https://github.com/spack/spack/pull/44210

**Comment by wdconinc** [comment]
Can you `git revert 4d168d0`? Then we can get back to merging this here.

**Review by wdconinc — APPROVED** [review]

**Pull request merged**